src/Altinn.Profile/Program.cs

```diff
@@ -96,29 +96,24 @@ async Task ConnectToKeyVaultAndSetApplicationInsights(ConfigurationManager confi
 {
     KeyVaultSettings keyVaultSettings = new();
     config.GetSection("kvSetting").Bind(keyVaultSettings);
-    if (!string.IsNullOrEmpty(keyVaultSettings.ClientId) &&
-        !string.IsNullOrEmpty(keyVaultSettings.TenantId) &&
-        !string.IsNullOrEmpty(keyVaultSettings.ClientSecret) &&
-        !string.IsNullOrEmpty(keyVaultSettings.SecretUri))
+    if (!string.IsNullOrEmpty(keyVaultSettings.SecretUri))
     {
-        logger.LogInformation("Program // Configure key vault client // App");
-        Environment.SetEnvironmentVariable("AZURE_CLIENT_ID", keyVaultSettings.ClientId);
-        Environment.SetEnvironmentVariable("AZURE_CLIENT_SECRET", keyVaultSettings.ClientSecret);
-        Environment.SetEnvironmentVariable("AZURE_TENANT_ID", keyVaultSettings.TenantId);
-        var azureCredentials = new DefaultAzureCredential();
+        logger.LogInformation("Program // Set app insights connection string // App");
 
-        config.AddAzureKeyVault(new Uri(keyVaultSettings.SecretUri), azureCredentials);
+        DefaultAzureCredential azureCredentials = new();
 
         SecretClient client = new(new Uri(keyVaultSettings.SecretUri), azureCredentials);
 
+        config.AddAzureKeyVault(new Uri(keyVaultSettings.SecretUri), azureCredentials);
+
         try
         {
             KeyVaultSecret keyVaultSecret = await client.GetSecretAsync(VaultApplicationInsightsKey);
             applicationInsightsConnectionString = string.Format("InstrumentationKey={0}", keyVaultSecret.Value);
         }
         catch (Exception vaultException)
         {
-            logger.LogError(vaultException, "Unable to read application insights key.");
+            logger.LogError(vaultException, $"Unable to read application insights key.");
```

Comment:
No need for string interpolation here. Should actually be avoided in log statemtents. Sorry for not catching that in the Storage changes.